core/src/main/java/nu/yona/server/subscriptions/service/BuddyDisconnectMessageDTO.java

```diff
@@ -106,7 +106,15 @@ private MessageActionDTO handleAction_Process(UserDTO actingUser, BuddyDisconnec
 		{
 			buddyService.removeBuddyAfterBuddyRemovedConnection(actingUser.getID(), messageEntity.getUserID());
 
+			updateMessageStatusAsProcessed(messageEntity);
```

Comment:
I'm not sure I fully get what you mean.
The HATEOAS pattern normally communicates the possible actions through the URLs, so when the message can be processed, the ``/process`` URL exists. Once it's processed, that URL is gone.

Embedding the resulting message in the action response DTO is a good idea, as it saves a round trip.